tools/generators/xcschemes/src/Generator/Generator.swift

```diff
@@ -54,7 +55,31 @@ struct Generator {
             targetsByKey: targetsByKey
         )
 
-        let schemeInfos = customSchemeInfos + automaticSchemeInfos
+        let autogenerationConfigArguments = try await AutogenerationConfigArguments.parse(
+            from: arguments.autogenerationConfigFile
+        )
+
+        let filteredAutomaticSchemeInfos = try automaticSchemeInfos.filter { scheme in
+            // Apply scheme auto-generation exclude patterns
+            for pattern in autogenerationConfigArguments.schemeNameExcludePatterns {
+                do {
+                    let regex = try Regex(pattern)
+                    let matches = scheme.name.matches(of: regex)
+
+                    if matches.count > 0 {
+                        return false
+                    }
+                } catch {
+                    throw PreconditionError(message: """
```

Comment:
Should this be a `UsageError`?

Reply:
Makes sense to me

Reply:
https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/3027/commits/896adadba4f2d783b9ce46a428ab87486d668a5e